app/src/main/java/com/example/fittracker/mvp/presenter/MainScreenPresenter.java

```diff
@@ -25,4 +40,43 @@ public void onBackPressed() {
     public void onLogOutPressed() {
         view.onLogOutPressed();
     }
+
+    @Override
+    public void checkLocationPermission() {
+        if (!view.locationPermissionGranted()) {
+            view.requestPermissions();
+        } else {
+            setLocationObject();
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    @Override
+    public void setLocationObject() {
+        model.setLocationObject(locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER));
+    }
+
+    public void init() {
+        locationManager = view.getLocationManager();
+        checkLocationPermission();
+        model.setWeatherService(WeatherGenerator.createService(WeatherService.class));
```

Comment:
you can pass this as a parameter in the model constructor from the activity

Reply:
Done